scrypt.go

```diff
@@ -0,0 +1,37 @@
+//go:build !cmd_go_bootstrap
+
+package openssl
+
+// #include "goopenssl.h"
+import "C"
+import (
+	"unsafe"
+)
+
```

Comment:
`EVP_PBE_scrypt` is supported since OpenSSL 3, and maybe not for all providers. Please implement the `SupportsSscrypt() bool` function. The implementation will be almost identical to https://github.com/golang-fips/openssl/blob/eb155dada337b0de53873aa42e89724ad59aa768/pbkdf2.go#L15, replacing `PBKDF2` with `SCRYPT` and returning `false` in the OpenSSL 1 case.

Then use this function in the new test cases.